src/sagemaker/local/local_session.py

```diff
@@ -0,0 +1,168 @@
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import datetime
+import logging
+import platform
+import time
+
+import urllib3
+from botocore.exceptions import ClientError
+
+from sagemaker.local.image import _SageMakerContainer
+from sagemaker.session import Session
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.WARNING)
+
+
+class LocalSagemakerClient(object):
+    """A SageMakerClient that implements the API calls locally.
+
+    Used for doing local training and hosting local endpoints. It still needs access to
+    a boto client to interact with S3 but it won't perform any SageMaker call.
+
+    Implements the methods with the same signature as the boto SageMakerClient.
+    """
+    def __init__(self, sagemaker_session=None):
+        """Initialize a LocalSageMakerClient.
+
+        Args:
+            sagemaker_session (sagemaker.session.Session): a session to use to read configurations
+                from, and use its boto client.
+        """
+        self.train_container = None
+        self.serve_container = None
+        self.sagemaker_session = sagemaker_session or LocalSession()
+        self.s3_model_artifacts = None
+        self.model_name = None
+        self.primary_container = None
+        self.role_arn = None
+        self.created_endpoint = False
+
+    def create_training_job(self, TrainingJobName, AlgorithmSpecification, RoleArn, InputDataConfig, OutputDataConfig,
+                            ResourceConfig, StoppingCondition, HyperParameters, Tags=None):
+
+        self.train_container = _SageMakerContainer(ResourceConfig['InstanceType'], ResourceConfig['InstanceCount'],
+                                                   AlgorithmSpecification['TrainingImage'], self.sagemaker_session)
+
+        for channel in InputDataConfig:
+            data_distribution = channel['DataSource']['S3DataSource']['S3DataDistributionType']
+            if data_distribution != 'FullyReplicated':
+                raise RuntimeError("DataDistribution: %s is not currently supported in Local Mode" %
+                                   data_distribution)
+
+        self.s3_model_artifacts = self.train_container.train(InputDataConfig, HyperParameters)
+
+    def describe_training_job(self, TrainingJobName):
+        """Describe a local traininig job.
+
+        Args:
+            TrainingJobName (str): Not used in this implmentation.
+
+        Returns: (dict) DescribeTrainingJob Response.
+
+        """
+        response = {'ResourceConfig': {'InstanceCount': self.train_container.instance_count},
+                    'TrainingJobStatus': 'Completed',
+                    'TrainingStartTime': datetime.datetime.now(),
+                    'TrainingEndTime': datetime.datetime.now(),
```

Comment:
This another thing that we need to review soon. The timing and status returned should comply with the docker container execution.